include/ginkgo/core/matrix/batch_csr.hpp

```diff
@@ -247,6 +247,77 @@ class BatchCsr
           row_ptrs_(exec, (size[0]) + 1)
     {}
 
+
+    /**
+     * Creates an BatchCsr matrix by duplicating a CSR matrix
+     *
+     * @param exec  Executor associated to the matrix
+     * @param num_batch_entries  the number of batches to be stored
+     * @param csr_mat  the csr matrix to be duplicated
+     */
+    BatchCsr(std::shared_ptr<const Executor> exec,
+             const size_type num_batch_entries,
+             const matrix::Csr<value_type, index_type> *csr_mat)
+        : EnableBatchLinOp<BatchCsr>(
+              exec, batch_dim<2>(num_batch_entries, csr_mat->get_size())),
+          values_(exec, csr_mat->get_num_stored_elements() * num_batch_entries),
+          col_idxs_(exec, csr_mat->get_num_stored_elements()),
+          row_ptrs_(exec, (csr_mat->get_size()[0]) + 1)
+    {
+        size_type offset = 0;
```

Comment:
Minor note: you could put this code block in a common templated function and just call that function in the two new constructors.

Reply:
Good point.